jib-core/src/main/java/com/google/cloud/tools/jib/api/Containerizer.java

```diff
@@ -217,13 +218,32 @@ public Containerizer setApplicationLayersCache(Path cacheDirectory) {
   }
 
   /**
-   * Sets the {@link EventHandlers} to handle events dispatched during Jib's execution.
+   * Adds the {@code eventConsumer} to handle the {@link JibEvent} with class {@code eventClass}.
+   * The order in which handlers are added is the order in which they are called when the event is
+   * dispatched.
    *
-   * @param eventHandlers the {@link EventHandlers}
+   * <p><b>Note: Implementations of {@code eventConsumer} must be thread-safe.</b>
+   *
+   * @param eventType the event type that {@code eventConsumer} should handle
+   * @param eventConsumer the event handler
+   * @param <E> the type of {@code eventClass}
+   * @return this
+   */
+  public <E extends JibEvent> Containerizer addEventHandler(
+      Class<E> eventType, Consumer<E> eventConsumer) {
```

Comment:
Good catch

Reply:
Hold on for this a moment. I remember there was some issue around that. I'll take a look again later.

Reply:
Never mind.

Unrelated, since `EventHandlers` is internal, how about removing `EventHandler.add(Consumer<JibEvent>)` that is more like a helper method? It's enough to have `add(eventType, eventConsumer)`.